pkg/controller/release/release_controller.go

```diff
@@ -382,9 +356,141 @@ func (c *Controller) scheduleRelease(rel *shipper.Release) (*shipper.Release, er
 	)
 	diff.Append(releaseutil.SetReleaseCondition(&rel.Status, *condition))
 
-	klog.V(4).Infof("Release %q has been successfully scheduled", controller.MetaKey(rel))
+	strategyPatches, trans, err = c.ensureReleaseState(relinfo)
+	patches = append(patches, strategyPatches...)
```

Comment:
`strategyPatches` can be `nil` if `err != nil`, so better move this append after the error check.